src/set_distances.jl

```diff
@@ -0,0 +1,183 @@
+
+function _correct_distance_to_set(d::Real)
+    return ifelse(d < 0, zero(d), d)
+end
+
+function _correct_distance_to_set(d::AbstractVector{T}) where {T <: Real}
+    return [_correct_distance_to_set(di) for di in d]
+end
+
+"""
+    distance_to_set(v, s)
+
+Compute the distance of a value to a set.
+For some vector-valued sets, can return a vector of distances.
+When `v ∈ s`, the distance is zero (or all individual distances are zero).
+
+Each set `S` implements `distance_to_set(v::T, s::S)` with `T` of appropriate type.
+"""
+function distance_to_set end
+
+distance_to_set(v::Real, s::LessThan) = _correct_distance_to_set(v - s.upper)
```

Comment:
We meant:

a) Move this to https://github.com/JuliaOpt/MathOptInterface.jl/blob/e022626b962be01d19fb9b9fe89c77990a0b96c8/src/sets.jl#L168
b) Write it as 
```Julia
distance_to_set(x::T, s::LessThan{T}) where {T} = max(zero(T), x - s.upper)
```

Reply:
`distance_to_set(x::T, s::LessThan{T})` this signature adds another constraint on types, not just that the types are comparable (which is enough to compute the distance), but equal

Reply:
I guess the type restriction isn't critical. I meant more the `max(zero(T), x - s.upper)` part.